app/controllers/public_health_controller.rb

```diff
@@ -100,7 +100,11 @@ def tab_counts
     # Get patients by workflow and tab
     patients = patients_by_type(workflow, tab)
 
-    render json: { total: patients.size }
+    # Cache tab sizes
+    cache_key = "#{current_user.jurisdiction.id}-#{workflow}-#{tab}"
+    total = Rails.cache.fetch(cache_key, expires_in: 5.minutes, race_condition_ttl: 30.seconds) { patients.size }
+
+    render json: { total: total }
```

Comment:
It would be good to include `patients_by_type` within the cache miss block instead of so the SQL is not getting run every time. This current code is still going to run `patients_by_type` each time.